doc/api/esm.md

```diff
@@ -1053,11 +1053,19 @@ and parent URL. The module specifier is the string in an `import` statement or
 `import()` expression, and the parent URL is the URL of the module that imported
 this one, or `undefined` if this is the main entry point for the application.
 
+The `conditions` property on the `context` is an array of conditions for
+[Conditional Exports][] that apply to this resolution request. They can be used
+for looking up conditional mappings elsewhere or to modify the list when calling
+the default resolution logic. Since `resolve` hooks have the ability to remove
+matching of conditions that ship with node like `node`, this can lead to files
+being executed that were meant for other JavaScript environments.
```

Comment:
Can we split this up into two paragraphs, with the second explicitly calling out the actionable part:

```suggestion
The `conditions` property on the `context` is an array of conditions for
[Conditional Exports][] that apply to this resolution request. They can be used
for looking up conditional mappings elsewhere or to modify the list when calling
the default resolution logic.

_The Node.js default conditions `"node"` and `"import"` will always be in the default
conditions list and **must** be passed through in the returned conditions list to
ensure Node.js-compatible resolution logic._
```

Reply:
Agreed on making it two paragraphs. But I think that 2nd paragraph is too strong. E.g. I would want to reserve the right to pass `require` in the future instead of `import`. And there should be be an if before the **must**, otherwise it sounds like an unqualified **must** (which it isn't). How about:

> The Node.js [default conditions listed above][link-here] will always be in the `context.conditions` list passed to the hook. _If the hook wants to ensure Node.js-compatible resolution logic, the conditions list **must** be passed through to the `defaultResolve` function._

Reply:
Looks great - just be specific about the condition list - `"all items from this default condition list must be passed through..."`.

Reply:
Updated with the new text. Less cursive because multiple lines in markdown with cursive are annoying to manage. Felt okay because it makes the **must** stand out more imo.